test/json/declarations/class_test.ts

```diff
@@ -0,0 +1,490 @@
+            ]
+          }],
+          members: []
+        }]
+      }));
+    });
+
+    it('supports extends', () => {
+      expectTranslateJSON('declare class X extends Y {}').to.equal(prettyStringify({
+        kind: ConvertedSyntaxKind.SourceFile,
+        fileName: 'demo/some/main.ts',
+        statements: [{
+          kind: ConvertedSyntaxKind.ClassDeclaration,
+          modifiers: [],
+          name: 'X',
+          heritageClauses: [{
+            kind: ConvertedSyntaxKind.HeritageClause,
+            keyword: 'extends',
+            types: [{kind: ConvertedSyntaxKind.ExpressionWithTypeArguments, expression: 'Y'}]
+          }],
+          members: []
+        }]
+      }));
+    });
+
+    it('supports classes with both heritage clauses', () => {
+      expectTranslateJSON('declare class W extends X implements Y, Z {}').to.equal(prettyStringify({
+        kind: ConvertedSyntaxKind.SourceFile,
+        fileName: 'demo/some/main.ts',
+        statements: [{
+          kind: ConvertedSyntaxKind.ClassDeclaration,
+          modifiers: [],
+          name: 'W',
+          heritageClauses: [
+            {
+              kind: ConvertedSyntaxKind.HeritageClause,
+              keyword: 'extends',
+              types: [{kind: ConvertedSyntaxKind.ExpressionWithTypeArguments, expression: 'X'}]
+            },
+            {
+              kind: ConvertedSyntaxKind.HeritageClause,
+              keyword: 'implements',
+              types: [
+                {kind: ConvertedSyntaxKind.ExpressionWithTypeArguments, expression: 'Y'},
+                {kind: ConvertedSyntaxKind.ExpressionWithTypeArguments, expression: 'Z'}
+              ]
+            }
+          ],
+          members: []
+        }]
+      }));
+    });
+  });
+
+  describe('type parameters', () => {
+    it('should handle basic cases', () => {
+      expectTranslateJSON('declare class X<T> {}').to.equal(prettyStringify({
+        kind: ConvertedSyntaxKind.SourceFile,
+        fileName: 'demo/some/main.ts',
+        statements: [{
+          kind: ConvertedSyntaxKind.ClassDeclaration,
+          modifiers: [],
+          name: 'X',
+          typeParameters: [{kind: ConvertedSyntaxKind.TypeParameter, name: 'T'}],
+          members: []
+        }]
+      }));
+
+      expectTranslateJSON('declare class X<T1, T2> {}').to.equal(prettyStringify({
+        kind: ConvertedSyntaxKind.SourceFile,
+        fileName: 'demo/some/main.ts',
+        statements: [{
+          kind: ConvertedSyntaxKind.ClassDeclaration,
+          modifiers: [],
+          name: 'X',
+          typeParameters: [
+            {kind: ConvertedSyntaxKind.TypeParameter, name: 'T1'},
+            {kind: ConvertedSyntaxKind.TypeParameter, name: 'T2'}
+          ],
+          members: []
+        }]
+      }));
+    });
+
+    it('should handle type parameters with constraints', () => {
+      expectTranslateJSON('declare class X<T extends string> {}').to.equal(prettyStringify({
```

Comment:
Can the script handle nested generics e.g. Foo<Bar extends Baz<Boo extends ...>>?

Reply:
Or I guess a better question: does it need to?

Reply:
It won't accept that syntax, but you can do something similar like
`interface X<U extends number, T extends Promise<U>> {}`

I'll add a test for this